features/ALS010_Alignment-segment-shape-representation-has-the-correct-number-of-items.feature

```diff
@@ -0,0 +1,12 @@
+@implementer-agreement
+@ALS
+@version1
+@E00040
+Feature: ALS010 - Alignment segment shape representation has the correct number of items
+
+The rule verifies that a shape representation of IfcAlignmentSegment has the correct number of items
+
+  Scenario: Agreement on shape representation of IfcAlignmentSegment having the correct number of items
+    Given A model with Schema "IFC4.3"
+    Given An IfcAlignmentSegment
+    Then A representation must have 2 items for PredefinedType of HELMERTCURVE and 1 item for all other values of PredefinedType
```

Comment:
I did consider it, I could not work out how to handle `=` versus `!=` as an operation.  I assumed I would need a scenario for every possible type which would have gotten quite verbose.

Is it as simple as passing the operator as a variable?  Something like:
```python
@gherkin_ifc.step("DesignParameters.PredefinedType {operator} 'HELMERTCURVE'")
def step_imp(context, inst, operator):
    if operator == "=":
          # logic for helmert curve type
    else:
          # logic for all other types
```

Reply:
I think this will be a good practical starting point to collect some improvements gathered over other pull requests and our work on the gcert. What about 

```feature
    Given A model with Schema "IFC4.3"
    Given An IfcAlignmentSegment
    Given Its attribute DesignParameters
    Given PredefinedType != HELMERTCURVE
    Given Return to IfcAlignmentSegment
    Given Its attribute Representation
    Given Its attribute Representations

    Then Lenght of attribute Items must be 1
```
Or, even better, 
```feature
... 
 Given Its attribute Items
 Then it must be of length 1
```
and something similar for `PredefinedType = HELMVERTCURVE,` but here the lenght must be 2.

There are three challenges here. Or, rather, points we've already partially discussed and solved. 

1.  `{attribute} != {value}` (PredefinedType != HELMVERTCURVE)
Just tested this in the branch of this PR and it appears to be solved with a seperate gherkin_ifc step and simply using `not` before the predicament. 
```python
    if hasattr(inst, attribute) and not pred(getattr(inst, attribute), value):
        yield ValidationOutcome(instance_id=inst, severity=OutcomeSeverity.PASSED)
```
2. Return to IfcAlignmentSegment
A similar case has been handled in the open PR for GEM006. I think this will be a good usecase to dig that up and add some modifications for it to function with all the new decorator functionalities. https://github.com/buildingSMART/ifc-gherkin-rules/blob/a0752ffb0049f3b81f747756573250053e2cc291/features/steps/steps/repeat_return.py#L17-L37

3. Lenght must be X ; specifiy expected value of inst
For instance, `inst` currently cannot be of type tuple or list, which makes this comparison difficult. We discussed this already with writing gcert, but we should be able to move back in the context stack or define the values we expect. This is a good opportunity to work on this. Something like 
`Then it must be of type tuple with length 1`

Reply:
Per discussion with @aothms we merged this PR as-is to prioritize the bugfix for ALS010 in v0.6.  These suggestions involve moving some of the logic from its current location as a custom implementation function into more standardized gherkin statements.  The net result is the same either way - the test suite returns the same results with the same error codes on the same instances.

We anticipate finding logical points in the future where the overall gherkin vocabulary could be reviewed and utilized to refactor custom implementations, maybe on an annual or semi-annual basis.